### DISCUSSION TITLE
20230907 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by josecelano** [nautilus-cyberneering/.github#57] [Meetings]
Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- CGarnham(@da2ce7)  - x
- JCelano(@josecelano) - x 
- CBosse(@cgbosse) - x
- MMelchor(@mario-nt) - x
- GByrne(@grmbyrn) - x

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by josecelano** [comment]
I've been working on the [Index Backend](https://github.com/torrust/torrust-index-backend).

### New sample repo to import torrents

I've finished importing all torrents from the https://academictorrents.com/. I've published the code I used to migrate data because It could be helpful for others to migrate/import data into the Turrust Index. The repo is:

https://github.com/torrust/torrents-importer-sample

I've also written a blog post about this new repo:

PR: https://github.com/torrust/torrust-website/pull/86

### Bug: Error trying to upload a torrent file [#266](https://github.com/torrust/torrust-index-backend/issues/266)

Issue: https://github.com/torrust/torrust-index-backend/issues/266

While doing the intensive testing (importing all torrents from https://academictorrents.com/), I found some torrents (6 out of more than 2400) that cannot be uploaded. Initially, I thought the problem was a new field in the bencoded torrent, but that was not the case. The problem is the function to parse the bencoded data is not working for those torrents. This is the function:

```rust
pub fn decode_torrent(bytes: &[u8]) -> Result<Torrent, Box<dyn error::Error>> {
    match de::from_bytes::<Torrent>(bytes) {
        Ok(torrent) => Ok(torrent),
        Err(e) => {
            println!("{e:?}");
            Err(e.into())
        }
    }
}
```

I tried to isolate the bug by creating a new project (a torrent file parser.):

https://github.com/torrust/torrust-parse-torrent

Surprisingly, the function is working there.

I've also added a [binary to the Index Backend](https://github.com/torrust/torrust-index-backend/pull/267) to parse torrents that could be helpful in the future to debug this type of error.

The bug is still not fixed. I've opened a PR on the package, adding a test for this case:

https://github.com/toby/serde-bencode/pull/32

The package is working fine. We must be using a different common dependency.

### Add support for multiple original infohashes [#261](https://github.com/torrust/torrust-index-backend/issues/261)

Issue: https://github.com/torrust/torrust-index-backend/issues/261

I'm adding a new table, as @da2ce7 proposed, with all the original infohashes for one canonical infohash. The canonical infohash is the resulting infohash after removing non-standard fields from the torrent `info` dictionary.

The main feature is done, but there are still some pending tasks:

https://github.com/torrust/torrust-index-backend/pull/269

**Comment by mario-nt** [comment]
### Torrust Frontend

- Working on https://github.com/torrust/torrust-index-frontend/pull/246
- Working on https://github.com/torrust/torrust-index-frontend/pull/251

### Torrust Backend

- Working on https://github.com/torrust/torrust-index-backend/pull/265

### Torrust types lib

- Closed https://github.com/torrust/torrust-index-types-lib/issues/9

### Torrust api lib

- Working on https://github.com/torrust/torrust-index-api-lib/issues/33

### Git

- Research on Conventional commits

### Training

- Semantic versioning 101
- Testing in Rust 101
- Rust (macros, test fixtures)

### Other

- Discussions with Jose about different topics

**Comment by josecelano** [comment]
I've created a [new discussion on the Index Backend to follow up the discussion we had about login the original torrent files](https://github.com/torrust/torrust-index-backend/discussions/272).

**Comment by grmbyrn** [comment]
## SvelteKit
* Learned more about two of the three core areas of SvelteKit: routing and loading.
* Wrote out notes in order to take advantage of spaced repetition, of which writing blog posts is a part.
* Learned about how to implement Markdown into SvelteKit, namely through packages such as mdsvex.

## Secure-Git-Guide
* Began implementing SvelteKit into project to replace Svelte which had been used up to this point.
* I had difficulty bringing in the markdown from the secure-git-guide articles due to problems with applying styles, errors from the code snippets, and accessing images and links.
* To do this, the articles were stored as an array of objects in a data.js file in an articles folder, which is inside a routes folder. The code below is a +page.server.js file which imports this data.js file and, in the load function, it maps over its objects, creating a new array of objects with specific properties (slug, title, and image) to be used as articles.

```
import {posts} from './data.js'

export function load(){
    return {
        articles: posts.map(post => ({
            slug: post.slug,
            title: post.title,
            image: post.image
        }))
    }
}
```
* Then, in the second code block below, the data.articles data is accessed in a +page.svelte file, also in the articles folder, to generate a list of blog articles with their titles and slugs for rendering in the SvelteKit component.

```
<script>
    export let data
</script>

<ul>
    {#each data.articles as {slug, title}}
        <li>
            <a href="/articles/{slug}">{title}</a>
        </li>
    {/each}
</ul>
```
* Each article from git-secure-guide is now available to be used in dynamic routing.

## Medium
* Wrote a couple of blog posts on routing in SvelteKit: https://medium.com/@grmbrn89/sveltekit-routing-407c5a2371bd and https://medium.com/@grmbrn89/ways-of-routing-630ec300f42e
